Libraries/LibC/sys/socket.h

```diff
@@ -135,4 +149,30 @@ int getpeername(int sockfd, struct sockaddr*, socklen_t*);
 int sendfd(int sockfd, int fd);
 int recvfd(int sockfd);
 
+// These three are non-POSIX, but common:
+#define CMSG_ALIGN(x) (((x) + sizeof(void*) - 1) & ~(sizeof(void*) - 1))
+#define CMSG_SPACE(x) (CMSG_ALIGN(sizeof(struct cmsghdr)) + CMSG_ALIGN(x))
+#define CMSG_LEN(x) (CMSG_ALIGN(sizeof(struct cmsghdr)) + (x))
+
+static inline struct cmsghdr* CMSG_FIRSTHDR(const struct msghdr* msg)
+{
+    if (msg->msg_controllen < sizeof(struct cmsghdr))
+        return 0;
```

Comment:
```suggestion
        return NULL;
```
perhaps?

Also are you sure it should be casting a `const` pointer to non-const pointer?

Reply:
Re const cast: I suppose it depends on if you picture the msghdr as a container of cmsghdrs (in which case you shouldn't hand out a mutable pointer since you can mutate part of the msghdr object through it), or as an object that's independent of the cmsghdrs and just points to them.

Semantically, it definitely would make sense to me if CMSG_NXTHDR took a const pointer in the 2nd argument and returned a mutable pointer -- the next cmsg isn't related to the first. But making that work would require const_cast<>s which we can't use in LibC headers.

I guess I'll just make all the args non-const, that's self-consistent and consistent with what most other unixes seem to do.